snuba/datasets/entities/spans.py

```diff
@@ -0,0 +1,78 @@
+from typing import Mapping, Sequence
+
+from snuba.clickhouse.columns import UUID, ColumnSet, DateTime, Nested
+from snuba.clickhouse.columns import SchemaModifiers as Modifiers
+from snuba.clickhouse.columns import String, UInt
+from snuba.clickhouse.translators.snuba.mappers import SubscriptableMapper
+from snuba.clickhouse.translators.snuba.mapping import TranslationMappers
+from snuba.datasets.entities import EntityKey
+from snuba.datasets.entity import Entity
+from snuba.datasets.plans.single_storage import SingleStorageQueryPlanBuilder
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_writable_storage
+from snuba.pipeline.simple_pipeline import SimplePipelineBuilder
+from snuba.query.data_source.join import ColumnEquivalence, JoinRelationship, JoinType
+from snuba.query.extensions import QueryExtension
+from snuba.query.processors import QueryProcessor
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+
+
+class SpansEntity(Entity):
+    def __init__(self) -> None:
+        storage = get_writable_storage(StorageKey.SPANS)
+
+        super().__init__(
+            storages=[storage],
+            query_pipeline_builder=SimplePipelineBuilder(
+                query_plan_builder=SingleStorageQueryPlanBuilder(
+                    storage=storage,
+                    mappers=TranslationMappers(
+                        subscriptables=[
+                            SubscriptableMapper(None, "tags", None, "tags")
+                        ],
+                    ),
+                ),
+            ),
+            abstract_column_set=ColumnSet(
+                [
+                    ("project_id", UInt(64)),
+                    ("transaction_id", UUID()),
+                    ("trace_id", UUID()),
+                    ("transaction_span_id", UInt(64)),
+                    ("span_id", UInt(64)),
+                    ("parent_span_id", UInt(64, Modifiers(nullable=True))),
+                    ("transaction_name", String()),
+                    ("op", String()),
+                    ("status", UInt(8)),
+                    ("start_ts", DateTime()),
+                    ("start_ns", UInt(32)),
+                    ("finish_ts", DateTime()),
+                    ("finish_ns", UInt(32)),
+                    ("duration_ms", UInt(32)),
+                    ("tags", Nested([("key", String()), ("value", String())])),
```

Comment:
Why does the list need to be redefined? Can't we use the same columns from the spans storage?

Reply:
There are fewer columns.
I removed description (for now), We do not want the tags hashmap and I did not expose partition and offset.